pkg/controller/openpitrix/helmrelease/helm_release_controller.go

```diff
@@ -183,12 +183,8 @@ func (r *ReconcileHelmRelease) reconcile(instance *v1alpha1.HelmRelease) (reconc
 		// no operation
 		return reconcile.Result{}, nil
 	case v1alpha1.HelmStatusFailed:
-		// Release used to be failed, but instance.Status.Version not equal to instance.Spec.Version
-		if instance.Status.Version > 0 && instance.Status.Version != instance.Spec.Version {
-			return r.createOrUpgradeHelmRelease(instance, true)
-		} else {
```

Comment:
The case of failed state is to do an update operation on the application in the failed state.

Reply:
I think this is still more standardized to be handled in the controller. Avoiding going into `if instance.Status.Version > 0 && instance.Status.Version != instance.Spec.Version` again would be fine, should just add updating the status to `HelmStatusUpgrading ` in there.

Reply:
It should also be necessary to update the state of all error handling in `createOrUpgradeHelmRelease()`, otherwise you'll get stuck in a dead loop even if you change the situation through the API.